src/components/print/PrintDirective.js

```diff
@@ -493,26 +331,41 @@ goog.require('ga_time_service');
           }
 
           var geometry = feature.getGeometry();
+          var styleToEncode;
+          if (styles && styles.length > 0) {
+            styleToEncode = styles[0];
+          }
 
-          // We encode the feature only if the geometry intersects the extent
-          if (geometry.intersectsExtent(getPrintRectangleCoords())) {
+          // We encode the feature only if the feature has a style.
+          if (!styleToEncode) {
+            return {
+              encFeatures: [],
+              encStyles: []
+            };
+          }
 
-            // Transform an ol.geom.Circle to a ol.geom.Polygon
-            if (geometry instanceof ol.geom.Circle) {
-              var polygon = gaPrintStyleService.olCircleToPolygon(geometry);
-              feature = new ol.Feature(polygon);
-            }
+          // Transform an ol.geom.Circle to a ol.geom.Polygon
+          if (geometry instanceof ol.geom.Circle) {
+            var polygon = gaPrintStyle.olCircleToPolygon(geometry);
+            feature = new ol.Feature(polygon);
+          }
 
-            // Handle ol.style.RegularShape by converting points to poylgons
-            var image = styles[0].getImage();
-            if (image instanceof ol.style.RegularShape) {
-              var scale = parseFloat($scope.scale.value);
-              var resolution = scale / UNITS_RATIO / POINTS_PER_INCH;
-              feature = gaPrintStyleService.olPointToPolygon(
-                  styles[0], feature, resolution);
-            }
+          // Handle ol.style.RegularShape by converting points to polygons
+          var image = styleToEncode.getImage();
+          if (image instanceof ol.style.RegularShape) {
+            var scale = parseFloat($scope.scale.value);
+            var resolution = scale / UNITS_RATIO / POINTS_PER_INCH;
+            feature = new ol.Feature(gaPrintStyle.olPointToPolygon(
+              feature.getGeometry(),
+              image.getRadius(),
+              resolution,
+              image.getPoints(),
+              image.getRotation()
+            ));
+          }
 
-            // Encode a feature
+          // Encode a feature if it intersects with the extent
+          if (geometry.intersectsExtent(getPrintRectangleCoords())) {
```

Comment:
before you should add:
geometry = feature.getGeometry();
as we may have change the feature in between.